### PR TITLE
New Scaffolding should not appear on Dependencies node. Fixes https:/…

### PR DESCRIPTION
**Pull request by nosami**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1024771

### CODE REVIEW

**Comment by nosami** [comment]
@monojenkins back port release-8.4

**Comment by rodrmoya** [comment]
@monojenkins backport release-8.4

File: main/src/addins/MonoDevelop.AspNetCore/MonoDevelop.AspNetCore.Commands/ScaffoldCommandHandler.cs [hunk 1]
```diff
@@ -49,8 +49,12 @@ public async void Scaffold ()
 		public void ScaffoldUpdate (CommandInfo info)
 		{
 			var project = CurrentNode.GetParentDataItem (typeof (DotNetProject), true) as DotNetProject;
-
-			info.Enabled = info.Visible = IsAspNetCoreProject (project);
+			info.Enabled = info.Visible = NodeIsFileOrFolder(CurrentNode) && IsAspNetCoreProject (project);
+		}
+
+		bool NodeIsFileOrFolder (ITreeNavigator node)
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.AspNetCore/MonoDevelop.AspNetCore.Commands/ScaffoldCommandHandler.cs]:
```suggestion
		static bool NodeIsFileOrFolder (ITreeNavigator node)
```

**Comment by nosami** [comment]
@monojenkins backport release-8.4

File: main/src/addins/MonoDevelop.AspNetCore/MonoDevelop.AspNetCore.Commands/ScaffoldCommandHandler.cs [hunk 2]
```diff
@@ -1,67 +1,67 @@
...
+		public override bool CanBuildNode (Type dataType)
+		{
+			return true;
+		}
+	}
+
+	class ScaffoldCommandHandler : NodeCommandHandler
+	{
+		[CommandHandler (AspNetCoreCommands.Scaffold)]
+		public async void Scaffold ()
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.AspNetCore/MonoDevelop.AspNetCore.Commands/ScaffoldCommandHandler.cs]: We don't need async here, I think.

**Comment by nosami** [comment]
> Can go in with or without removing async.

The build appears to have hung, so may as well make the change now.

**Comment by nosami** [comment]
@monojenkins backport release-8.4

**Pull request merged**